// react-native-profile-header/build/dist/ProfileHeader.style.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports._container = void 0;
const react_native_1 = require("react-native");
exports._container = (height, backgroundColor) => ({
    width: "100%",
    height,
    top: 0,
    paddingLeft: 24,
    paddingRight: 24,
    backgroundColor,
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "center",
});
exports.default = react_native_1.StyleSheet.create({
    container: {},
    leftAlignedContainer: {
        marginRight: "auto",
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
    },
    leftAlignedButtonImageStyle: {
        height: 25,
        width: 25,
    },
    titleContainer: {
        marginLeft: 16,
    },
    titleTextStyle: {
        fontSize: 18,
        fontWeight: "bold",
    },
    rightAlignedContainer: {
        marginLeft: "auto",
        flexDirection: "row",
        alignItems: "center",
        justifyContent: "center",
    },
    iconButtonContainer: {
        marginRight: 24,
    },
    iconImageStyle: {
        width: 18,
        height: 18,
    },
    profileImageStyle: {
        width: 50,
        height: 50,
        borderRadius: 12,
    },
});
